neo/Ledger/MemoryPool.cs

```diff
@@ -253,6 +255,34 @@ public IEnumerable<Transaction> GetSortedVerifiedTransactions()
             }
         }
 
+        // This function return some statistics about the best _maxTxPerBlock and _maxLowPriorityTxPerBlock txs of the mempool
+        public JObject GetMemPoolInfoForNextBlock()
+        {
+            int minSizeSortedHP = Math.Min(_maxTxPerBlock, _sortedHighPrioTransactions.Count);
+            int minSizeSortedLP = Math.Min(_maxLowPriorityTxPerBlock, _sortedLowPrioTransactions.Count);
+            int minSizeUnverifiedHP = Math.Max(Math.Min(_maxTxPerBlock, _unverifiedSortedHighPriorityTransactions.Count) - minSizeSortedHP, 0);
+            int minSizeUnverifiedLP = Math.Max(Math.Min(_maxLowPriorityTxPerBlock, _unverifiedSortedLowPriorityTransactions.Count) - minSizeSortedLP, 0);
+                                      
+            JObject json = new JObject();
+            _txRwLock.EnterReadLock();
+            try
+            {
+                json["sortedHP_avgNetFees"] = new JArray((_sortedHighPrioTransactions.Reverse().Take(minSizeSortedHP).Sum(p => p.Tx.NetworkFee) / Math.Max(minSizeSortedHP, 1)).ToString());
```

Comment:
I think the sums should be calculated in memory as transactions are added / removed from the pool instead of iterating them all here to get the sums. This will work also if you make the change I suggested above about always using all transactions instead of how many this node thinks will be verified each block since CN nodes policy could be different and not even the same for all of them.

Reply:
I like the idea of the SUM in cache, as an Auxiliary Data Structure.
We just need to be sure that they are not used when RPC is not enabled.